auth/generate-google-id-jwt.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Python script generates a Google ID token based on the input payload"""
+
+import base64
+import httplib
+import json
+import argparse
+import time
+import urllib
+
+import oauth2client.crypt
+import googleapiclient.discovery
+from oauth2client.service_account import ServiceAccountCredentials
+from oauth2client.client import GoogleCredentials
+
+def generate_jwt(args):
+    """Generates a signed JSON Web Token using a service account. Based on https://cloud.google.com/endpoints/docs/service-to-service-auth"""
+    # Make sure the service account has "Service Account Token Creator" permissions in Google IAM
+    credentials = ServiceAccountCredentials.from_json_keyfile_name(
+      args.service_account_file).create_scoped(['https://www.googleapis.com/auth/cloud-platform'])
```

Comment:
You can mention that you use "cloud-platform" as an example scope, but the service account can be assigned any scope.